Source/WebCore/dom/Attr.cpp

```diff
@@ -101,9 +102,31 @@ ExceptionOr<void> Attr::setPrefix(const AtomString& prefix)
 
 ExceptionOr<void> Attr::setValue(const AtomString& value)
 {
-    if (RefPtr element = m_element.get())
-        return element->setAttribute(qualifiedName(), value, true);
-    else
+    if (RefPtr element = m_element.get()) {
+        auto verifiedValue = value;
+        if (document().requiresTrustedTypes()) {
+            AttributeTypeAndSink attributeTypeAndSink = trustedTypeForAttribute(element->nodeName(),
+            qualifiedName().localName(), element->namespaceURI(), qualifiedName().namespaceURI());
+            if (!attributeTypeAndSink.attributeType.isNull()) {
+                auto attributeValue = trustedTypesCompliantAttributeValue(*document().scriptExecutionContext(),
+                    attributeTypeAndSink.attributeType, value, attributeTypeAndSink.sink);
+                if (attributeValue.hasException())
+                    return attributeValue.releaseException();
+
+                verifiedValue = AtomString { attributeValue.releaseReturnValue() };
+
+                if (!m_element.get()) {
+                    m_standaloneValue = verifiedValue;
+                    return { };
+                }
+
+                if (RefPtr newElement = m_element.get(); newElement != element)
+                    return { };
```

Comment:
I suggest this refactor.

But it makes me doubt whether we have the `m_standaloneValue` logic exactly. Is it correct to drop the new value on the floor entirely and not raise an exception when the element underlying the `Attr` changes to a non-null value, but record the new value as a standalone value if the element changes to a null value? Do we have test coverage for these cases? To construct those tests, wondering how the trusted type calls can lead to run code that can end up changing the value of `m_element`? Also, is there some way we can share the `m_standaloneValue` logic so it’s not repeated twice?

By the way, since we never do anything with `newElement` other than compare it with `element` and check it for null, this _might_ be a rare case where we don’t need a `RefPtr`.

```suggestion
                if (RefPtr newElement = m_element.get(); newElement != element) {
                    if (!newElement)
                        m_standaloneValue = WTFMove(verifiedValue);
                    return { };
                }
```

Reply:
I've done the suggested refactor, I have kept RefPtr for now, I'm not sure how to easily determine if not using it is okay?

Regarding whether this is behaving correctly, per the current spec PR it is, though Anne did raise that it might be better to throw: https://github.com/whatwg/dom/pull/1268#discussion_r2207267384

Regarding test coverage, yes we do, it's in https://wpt.fyi/results/trusted-types/set-attributes-mutations-in-callback.tentative.html - WebKit currently has a number of failures and this change resolves them all.

Reply:
> I have kept `RefPtr` for now, I'm not sure how to easily determine if not using it is okay?

Well, we know it’s OK; comparing with another value doesn’t require `RefPtr`. The question is whether the Safer CPP analysis will flag it as not OK. We should not use `RefPtr` and see if the the bot complains.

Reply:
Just to confirm it should be like below?

```
if (auto* newElement = m_element.get(); newElement != element) {
    if (!newElement)
        m_standaloneValue = WTFMove(verifiedValue);
    return { };
}
```

Reply:
Looks like Safer CPP is complaining about it, so just put the `RefPtr` back, please. @rniwa, I’m really surprised that it doesn’t understand this is safe.